kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java

```diff
@@ -295,6 +313,77 @@ private void decryptAndConvertPasswordProperties(Map<String, Object> modifiedPro
         }
     }
 
+    private void decryptAndConvertCertificatesProperties(Map<String, Object> modifiedProps) throws KuraException {
+
+        for (Entry<String, Object> prop : modifiedProps.entrySet()) {
+            if (prop.getKey().contains("802-1x.keystore.pid")) {
+                String keystorePid = (String) prop.getValue();
+
+                String interfaceName = prop.getKey().split("\\.")[2];
+                findAndDecodeCertificatesForInterface(interfaceName, modifiedProps, getKeystore(keystorePid));
```

Comment:
Final thought: do you really need this? Do you really need to know the interface name to decrypt the certificate? Can't you just iterate over the properties and, if you find one that matches `.config.802-1x.client-cert-name` (regardless of the interface name) decrypt it? The same goes for the other properties.

... or am I missing something?